common/perf-test-core/src/main/java/com/azure/perf/test/core/MockEventProcessor.java

```diff
@@ -0,0 +1,147 @@
+package com.azure.perf.test.core;
+
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+import java.util.stream.IntStream;
+
+/**
+ * Represents a Mock event processor.
+ */
+public class MockEventProcessor {
```

Comment:
Should this be in sdk/eventhubs directory as event processor is specific to Event Hubs. Service Bus does have a processor too but they don't have partitions like this one has.

Reply:
We can have a copy there, if needed. 
Else we'll need to depend on EH SDK in perf core.

Reply:
The `MockEventProcessor` exists to test the event-based functionality in the perf framework without depending on a real SDK like Event Hubs.  So this class should be in the same folder used for other sample perf tests.  In Java, I believe this is currently the same folder as the perf framework itself, but it should eventually be moved into a separate project (https://github.com/Azure/azure-sdk-for-java/issues/18737).